open_fdd/ecm_engineering/workbook.py

```
@@ -0,0 +1,133 @@
+        return dict(self.model["modules"][module]["cells"])
+
+    def _api_ref(self, api_key: str) -> tuple[str, str]:
+        if api_key in self.model.get("global_inputs", {}):
+            return tuple(self.model["global_inputs"][api_key].split("!", 1))  # type: ignore[return-value]
+        for module in self.model["modules"].values():
+            if api_key in module["cells"]:
+                return module["sheet"], module["cells"][api_key]
+        raise KeyError(f"unknown API key: {api_key}")
+
+    def _sheet_paths(self, zin: zipfile.ZipFile) -> dict[str, str]:
+        workbook = ET.fromstring(zin.read("xl/workbook.xml"))
```

Ruff (0.15.21)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)